src/presentation/trust_anchor.rs

```diff
@@ -0,0 +1,546 @@
+        return Err(Error::MdocAuth(
+            "The root certificate is expected to contain CA=true in the Basic Constraints, but found an empty sequence".to_string()
+        ));
+    };
+
+    if !ca.as_boolean()?.bool() {
+        errors.push(Error::MdocAuth(format!("The root certificate is expected to contain Basic Constraints CA=true, but found: {:?}", ca)));
+    }
+    let Some(path_len) = iter.next() else {
+        return Err(Error::MdocAuth("The root certificate is expected to contain pathLen:0 in the Basic Constraints, but it was not found".to_string()));
+    };
+
+    if [path_len.as_integer()?.as_u32()?] != BASIC_CONSTRAINTS {
+        errors.push(Error::MdocAuth(format!("The root certificate is expected to contain Basic Constraints pathLen=0, but found: {:?}", path_len)));
+    }
+
+    //CRL Distribution Points  2.5.29.31
+    let Some(crl_distribution_point) = root_extensions
+        .iter()
+        .find(|ext| ext.extn_id.to_string() == *OID_CRL_DISTRIBUTION_POINTS)
+    else {
+        return Err(Error::MdocAuth("The root certificate is expected to have a crl distribution point specificied, but the extensions was not found".to_string()));
+    };
+
+    let crl_dp: (_, SequenceOf<Any>) =
+        FromDer::from_der(crl_distribution_point.extn_value.as_bytes())
+            .map_err(|e| Error::MdocAuth(e.to_string()))?;
+    let Some(distribution_points) = crl_dp.1.iter().next() else {
+        return Err(Error::MdocAuth(
+            "The root certificate is expected to have a crl distribution point specificied, but the extension value was not found"
+            .to_string(),
+        ));
+    };
+    let dp: (_, Any) =
+        FromDer::from_der(distribution_points.data).map_err(|e| Error::MdocAuth(e.to_string()))?;
+    if dp.1.tag().0 != 0 {
+        errors.push(Error::MdocAuth(
+            "reason and crlIssuer fields shall not be used in the crl distribution point"
+                .to_string(),
+        ));
+    }
+
+    // Issuer Alternative Name  2.5.29.18
+    let Some(issuer_alternative_name) = root_extensions
+        .iter()
+        .find(|ext| ext.extn_id.to_string() == *OID_ISSUER_ALTERNATIVE_NAME)
+    else {
+        return Err(Error::MdocAuth(
+            "The root certificate is expected to have issuer alternative name specificied, but the extensions was not found".to_string()
+        ));
+    };
+
+    let ian: (_, SequenceOf<Any>) =
+        FromDer::from_der(issuer_alternative_name.extn_value.as_bytes())
+            .map_err(|e| Error::MdocAuth(e.to_string()))?;
+    for item in ian.1.iter() {
+        if item.tag().0 != 1 && item.tag().0 != 6 {
+            errors.push(Error::MdocAuth(format!(
+                "issuer alternative name is expected to be an rfc822name or a URI, but found: {:?}",
+                item
+            )));
+        }
+    }
+
+    Ok(errors)
+}
+
+pub fn validate_iaca_signer_extensions(
+    leaf_extensions: Vec<Extension>,
+) -> Result<Vec<Error>, Error> {
```

Comment:
A lot of this code could be simplified by using the Extension types already defined in the x509-cert crate. Along the lines of:

```rust
use x509_cert::ext::pkix::KeyUsage;
let key_usage: KeyUsage = leaf_extensions.iter().find(|extension| extension.extn_id == <KeyUsage as AssociatedOid>::OID).unwrap();
```